src/components/HistoryRecord/HistoryRecord.tsx

```diff
@@ -314,3 +343,110 @@ export const HistoryRecordParticipant: React.FC<HistoryChangeProps<Activity & {
         }
     />
 );
+
+interface HistoryRecordCriteriaProps {
+    title: string;
+    strike: boolean;
+    isChangeAction?: boolean;
+}
+
+const HistoryRecordSimplifyCriteria: React.FC<HistoryRecordCriteriaProps & { state: boolean }> = ({
+    title,
+    state,
+    isChangeAction,
+}) => {
+    return (
+        <>
+            <Text size="xs" weight="bold">
+                {title}
+            </Text>
+            {nullable(isChangeAction, () => (
+                <Text size="xs">{tr(state ? 'as completed' : 'as uncompleted')}</Text>
+            ))}
+        </>
+    );
+};
+
+const HistoryRecordGoalCriteria: React.FC<HistoryRecordCriteriaProps & { shortId: string; id: string }> = ({
+    title,
+    shortId,
+    id,
+    strike,
+}) => {
+    return (
+        <StyledIssueListItem
+            strike={strike}
+            size="xs"
+            issue={{
+                title,
+                _shortId: shortId,
+                id,
+            }}
+        />
+    );
+};
+
+export const HistoryRecordCriteria: React.FC<
+    HistoryChangeProps<GoalAchieveCriteria & { goalAsCriteria: Goal & { state: StateData | null } }> & {
+        action: HistoryAction;
+    }
+> = ({ to, action }) => {
+    const recordCtx = useContext(RecordCtx);
+
+    const isChangeAction = ['complete', 'uncomplete'].includes(action);
+
+    recordCtx.setSubjectText((prev) => {
+        if (to?.goalAsCriteria != null) {
+            if (isChangeAction) {
+                if (action === 'complete') {
+                    return 'goalComplete';
+                }
+                return 'goalInProgress';
+            }
+            return 'goalAsCriteria';
+        }
+
+        if (isChangeAction) {
+            return 'criteriaState';
+        }
+
+        return prev;
+    });
+
+    if (to?.goalAsCriteria) {
+        return (
+            <HistorySimplifyRecord
+                withPretext={false}
+                to={
+                    <>
+                        <HistoryRecordGoalCriteria
+                            title={to.goalAsCriteria.title}
+                            shortId={`${to.goalAsCriteria.projectId}-${to.goalAsCriteria.scopeId}`}
+                            id={to.goalAsCriteria.id}
+                            strike={action === 'remove'}
+                        />
+                        <Text size="xs">{tr('as criteria')}</Text>
+                    </>
+                }
+            />
+        );
+    }
+
+    if (to) {
+        return (
+            <HistorySimplifyRecord
+                withPretext={false}
+                to={
+                    <HistoryRecordSimplifyCriteria
+                        title={to.title}
+                        state={isChangeAction && action === 'complete'}
+                        strike={action === 'remove'}
+                        isChangeAction={isChangeAction}
```

Comment:
Why not to do this same as on [428](https://github.com/taskany-inc/issues/pull/1294/files#diff-28e7bb86d913c391369eee9c1f6a73620a9b73b57ac0771d78c2eff181931ac4R428) line? It's better in terms of composition.